### DISCUSSION TITLE
Using an access token for (user) data for showing first name and avatar

### DISCUSSION THREAD

**Discussion by vanling** [Intevel/nuxt-directus#27] [Q&A] [answered]
Say i want to get blog items including the user first_name + user avatar who created the blog item? I don't want to open up all the user data to public. 

There are many ways to look at it.. I currently created a user with a token with access to the firstname and avatar source only. So in nuxt i do a `$fetch(url?access_token=)` for blog items.. this way still opens all first names and avatar to public. 

Is it possible with nuxt-directus to use an access token next to/instead of the current auth flow? 
How would you guys would solve 'a problem' like this?

**Comment by Intevel** [comment, marked as answer]
The module has several composables, an option to implement the feature wouldn't be too bad. 

All composables that use data fetched from the directus server use the `useDirectus` composable (which also uses `$fetch`). You can assign parameters like `access_token` to the `useDirectus` composable. However, assign an `access_token` for e.g. the `useDirectusItems` composable is **currently** not implemented. Would be a good feature though. 🤟
